packages/nimble-components/src/wafer-map/modules/wafer-map-update-tracker.ts

```diff
@@ -93,16 +93,27 @@ export class WaferMapUpdateTracker extends UpdateTracker<typeof trackedItems> {
      * After the update is finished, all the tracked items are reset.
      */
     public override queueUpdate(): void {
-        if (!this.wafermap.$fastController.isConnected) {
+        if (!this.wafermap.$fastController.isConnected || this.updateQueued) {
             return;
         }
-        if (!this.updateQueued) {
-            this.updateQueued = true;
+        this.updateQueued = true;
+        if (this.wafermap.currentTask === undefined) {
             DOM.queueUpdate(() => {
                 this.wafermap.update();
                 this.untrackAll();
                 this.updateQueued = false;
             });
+        } else {
+            this.wafermap.currentTask.then(
+                () => {
+                    DOM.queueUpdate(() => {
+                        this.wafermap.update();
+                        this.untrackAll();
+                        this.updateQueued = false;
+                    });
+                },
+                () => {}
+            );
```

Comment:
Rewrite this to avoid `.then` and preefer async/await

something like
```ts
void async () => {
    await this.wafermap.currentTask;
    DOM.queueUpdate(() => {
        this.wafermap.update();
        this.untrackAll();
        this.updateQueued = false;
    });
}();
```

Reply:
sure! solved here https://github.com/ni/nimble/pull/1929/commits/2acda5dfbd4a393c9553f0e0a9f06d78d7ee0baf